kernel/examples/inspect-table/src/main.rs

```diff
@@ -43,39 +46,161 @@ enum Commands {
     },
 }
 
-fn main() {
-    let cli = Cli::parse();
-    let path = std::fs::canonicalize(PathBuf::from(cli.path));
-    let Ok(path) = path else {
-        println!("Couldn't open table: {}", path.err().unwrap());
-        return;
-    };
-    let Ok(url) = url::Url::from_directory_path(path) else {
-        println!("Invalid url");
-        return;
+fn main() -> ExitCode {
+    env_logger::init();
+    match try_main() {
+        Ok(()) => ExitCode::SUCCESS,
+        Err(e) => {
+            println!("{e:#?}");
+            ExitCode::FAILURE
+        }
+    }
+}
+
+enum Action {
+    Metadata(delta_kernel::actions::Metadata, usize),
+    Protocol(delta_kernel::actions::Protocol, usize),
+    Remove(delta_kernel::actions::Remove, usize),
+    Add(delta_kernel::actions::Add, usize),
+}
+
+impl Action {
+    fn row(&self) -> usize {
+        match self {
+            Action::Metadata(_, row) => *row,
+            Action::Protocol(_, row) => *row,
+            Action::Remove(_, row) => *row,
+            Action::Add(_, row) => *row,
+        }
+    }
+}
+
+fn fields_in(field: &StructField) -> usize {
+    if let DataType::Struct(ref inner) = field.data_type {
+        let mut count = 0;
```

Comment:
nit: Not sure if we have a preference for FP vs imperative, but we could do this 
```rust 
        inner.fields().map(fields_in).sum()
```